pkg/provisioning/validate.go

```diff
@@ -71,7 +71,10 @@ func validateProcessorsConfig(mp map[string]ProcessorConfig) error {
 	var err error
 	for k, cfg := range mp {
 		if cfg.Type == "" {
-			err = multierror.Append(err, cerrors.Errorf("processors %q: \"type\" is mandatory: %w", k, ErrMandatoryField))
+			err = multierror.Append(err, cerrors.Errorf("processor %q: \"type\" is mandatory: %w", k, ErrMandatoryField))
+		}
+		if cfg.Workers < 0 {
+			err = multierror.Append(err, cerrors.Errorf("processor %q: \"workers\" can't be negative: %w", k, ErrInvalidField))
```

Comment:
Hmm, I'm not sure, I have a few concerns:
- We would duplicate the code that's already in the SDK.
- We could only partially validate the SDK based on the specifications. In other words - the config is not necessarily "valid" if the spec validations succeed, a connector is allowed to do manual validations in `Configure`.

And another more general concern about the way we validate the config - I'm not sure the behavior would be expected in case an existing pipeline is being updated through the configuration file. Currently, we simply don't apply the configuration if it's invalid. This means that Conduit keeps the old pipeline in memory which doesn't match the provided configuration. That seems strange, I would expect that Conduit would store the _invalid_ configuration in memory and just refuse to run the pipeline. So essentially I'm inclined to do even less validation rather than more.